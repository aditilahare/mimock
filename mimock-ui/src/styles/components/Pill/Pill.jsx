import React from 'react';
import PropTypes from 'prop-types';
import { PillWrapper } from './Pill.style';

const pillData = {
	RED_PILL: {
		background: 'bg-red-100',
		color: 'text-red-500',
		border: 'border-red-500',
	},
	GREEN_PILL: {
		background: 'bg-green-100',
		color: 'text-green-500',
		border: 'border-green-500',
	},
	YELLOW_PILL: {
		background: 'bg-yellow-100',
		color: 'text-yellow-500',
		border: 'border-yellow-500',
	},
	DEFAULT_PILL: {
		background: 'bg-white',
		color: 'text-gray-500',
		border: 'border-gray-500',
	},
};

function Pill({ variant, label }) {
	return (
		<PillWrapper data-testid='pill' variantprops={getVariant(variant)}>
			{label}
		</PillWrapper>
	);
}

function getVariant(variant) {
	return !variant ? pillData.DEFAULT_PILL : pillData[variant];
}

Pill.propTypes = {
	label: PropTypes.string.isRequired,
	variant: PropTypes.string,
};

export default Pill;
