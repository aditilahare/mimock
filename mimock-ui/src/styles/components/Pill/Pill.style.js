import tw from 'tailwind-styled-components';

export const PillWrapper = tw.div`
	px-6
	py-3
	m-auto
	shadow-md
	rounded-full
	text-sm
	text-center
	w-fit
	border-2
	cursor-pointer
	${(props) => Object.values(props.variantprops)}
`;
