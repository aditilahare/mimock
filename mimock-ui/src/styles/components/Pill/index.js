import Pill from './Pill.jsx';

export const PillVariants = {
	RedPill: 'RED_PILL',
	GreenPill: 'GREEN_PILL',
	YellowPill: 'YELLOW_PILL',
	DefaultPill: 'DEFAULT_PILL',
};

export default Pill;
