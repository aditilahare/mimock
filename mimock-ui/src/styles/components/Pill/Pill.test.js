import React from 'react';
import { render } from '@testing-library/react';
import Pill from './Pill';
import { PillVariants } from './index';

describe('Button', () => {
	let tree;

	it('should render red pill', async () => {
		tree = await render(<Pill variant={PillVariants.RedPill} label='Cancel' />);

		const { getByTestId, container } = tree;

		expect(getByTestId('pill')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('should render green pill', async () => {
		tree = await render(
			<Pill variant={PillVariants.GreenPill} label='Success' />
		);

		const { getByTestId, container } = tree;

		expect(getByTestId('pill')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('should render yellow pill', async () => {
		tree = await render(
			<Pill variant={PillVariants.YellowPill} label='Warning' />
		);

		const { getByTestId, container } = tree;

		expect(getByTestId('pill')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('should render default pill without variant', async () => {
		tree = await render(<Pill label='Default' />);

		const { getByTestId, container } = tree;

		expect(getByTestId('pill')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
