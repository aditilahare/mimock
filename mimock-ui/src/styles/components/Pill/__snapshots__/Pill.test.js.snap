// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render default pill without variant 1`] = `
<div>
  <div
    class="px-6 py-3 m-auto shadow-md rounded-full text-sm text-center w-fit border-2 cursor-pointer bg-white text-gray-500 border-gray-500"
    data-testid="pill"
    variantprops="[object Object]"
  >
    Default
  </div>
</div>
`;

exports[`Button should render green pill 1`] = `
<div>
  <div
    class="px-6 py-3 m-auto shadow-md rounded-full text-sm text-center w-fit border-2 cursor-pointer bg-green-100 text-green-500 border-green-500"
    data-testid="pill"
    variantprops="[object Object]"
  >
    Success
  </div>
</div>
`;

exports[`Button should render red pill 1`] = `
<div>
  <div
    class="px-6 py-3 m-auto shadow-md rounded-full text-sm text-center w-fit border-2 cursor-pointer bg-red-100 text-red-500 border-red-500"
    data-testid="pill"
    variantprops="[object Object]"
  >
    Cancel
  </div>
</div>
`;

exports[`Button should render yellow pill 1`] = `
<div>
  <div
    class="px-6 py-3 m-auto shadow-md rounded-full text-sm text-center w-fit border-2 cursor-pointer bg-yellow-100 text-yellow-500 border-yellow-500"
    data-testid="pill"
    variantprops="[object Object]"
  >
    Warning
  </div>
</div>
`;
